sdk/tests/conformance/offscreencanvas/context-lost-worker.html

```diff
@@ -0,0 +1,444 @@
+
+    // We need to initialize |uniformLocation| before losing context.
+    // Otherwise gl.getUniform() when context is lost will throw.
+    uniformLocation = gl.getUniformLocation(program, "tex");
+    loseContext();
+}
+
+function loseContext()
+{
+    // Note: this will cause the context to be lost, but the
+    // webglcontextlost event listener to be queued.
+    extension.loseContext();
+}
+
+function testValidContext()
+{
+    if (gl.isContextLost())
+        return false;
+
+    arrayBuffer = new ArrayBuffer(4);
+    arrayBufferView = new Int8Array(arrayBuffer);
+
+    // Generate resources for testing.
+    buffer = gl.createBuffer();
+    gl.bindBuffer(gl.ARRAY_BUFFER, buffer);
+    framebuffer = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, framebuffer);
+    program = setupSimpleTextureProgram(gl);
+    renderbuffer = gl.createRenderbuffer();
+    gl.bindRenderbuffer(gl.RENDERBUFFER, renderbuffer);
+    shader = gl.createShader(gl.VERTEX_SHADER);
+    texture = gl.createTexture();
+    gl.bindTexture(gl.TEXTURE_2D, texture);
+    if (gl.getError() != gl.NO_ERROR)
+        return false;
+
+    // Test is queries that will later be false
+    if (!compareGLError(gl.NO_ERROR, "gl.enable(gl.BLEND)"))
+        return false;
+    if (!gl.isBuffer(buffer) || !gl.isEnabled(gl.BLEND) || !gl.isFramebuffer(framebuffer) ||
+        !gl.isProgram(program) || !gl.isRenderbuffer(renderbuffer) || !gl.isShader(shader) ||
+        !gl.isTexture(texture))
+        return false;
+
+    if (OES_vertex_array_object) {
+        vertexArrayObject = OES_vertex_array_object.createVertexArrayOES();
+        if (gl.getError() != gl.NO_ERROR)
+            return false;
+        if (!OES_vertex_array_object.isVertexArrayOES(vertexArrayObject))
+            return false;
+    }
+    return true;
+}
+
+function testLostContext()
+{
+    // Functions with special return values.
+    if (!gl.isContextLost())
+        self.postMessage("Test failed");
+
+    if (gl.getError() != gl.CONTEXT_LOST_WEBGL)
+        self.postMessage("Test failed");
+    if (gl.getError() != gl.NO_ERROR)
+        self.postMessage("Test failed");
+
+    if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) != gl.FRAMEBUFFER_UNSUPPORTED ||
+        gl.getAttribLocation(program, 'u_modelViewProjMatrix') != -1 ||
+        gl.getVertexAttribOffset(0, gl.VERTEX_ATTRIB_ARRAY_POINTER) != 0)
+        self.postMessage("Test failed");
+
+    // Test the extension itself.
+    if (!compareGLError(gl.INVALID_OPERATION, "extension.loseContext()"))
+        self.postMessage("Test failed");
+
+    imageData = new ImageData(1, 1);
+    float32array = new Float32Array(1);
+    int32array = new Int32Array(1);
+
+    // Functions returning void should return immediately.
+    // This is untestable, but we can at least be sure they cause no errors
+    // and the codepaths are exercised.
+    if (!compareGLError(gl.NO_ERROR, "gl.activeTexture(gl.TEXTURE0)") ||
```

Comment:
This is an awful lot of duplicated code compared to the similar test for canvases on the main thread. Can you please figure out how to share the code? In the past, to share code between WebGL 1.0 and 2.0 tests, we refactored the JavaScript portions of the tests into sdk/tests/js/tests/ and had the HTML files refer to that code. Can the similar pattern be done to share the code for regular canvas tests and OffscreenCanvas?